### ISSUE TITLE
Missing audio message beginning for the very first publication in the room

### ISSUE THREAD

**Issue by denis-obukhov** [livekit/livekit#2827]
**Describe the bug**
I have a room with two participants. Autosubscribe is turned on for both sides. When a participant streams an audio publication, the listener often misses the first 1-2 seconds of the stream. I have observed this behavior in two combinations: iOS Client + iOS Client and iOS Client + GO SDK streaming a prerecorded audio.
Subsequent publications work fine, even after reconnection to the room.
Does LiveKit guarantee that an incoming publication is played from the very beginning? I suspect initial negotiations cut some time.

**Server**
- Version: [1.7.0]
- Environment: Local Ubuntu
- any other information about your deployment setup

**Client**
- SDK: iOS v2.0.11, Go v2.2.0

**To Reproduce**
1. two clients are connected to room
2. 1st begins transmission
3. 2nd listens to and misses audio beginning for the very first publication (intermittent issue) 

**Expected behavior**
All listeners are able to hear full audio message

**Comment by davidzhao** [comment]
> Does LiveKit guarantee that an incoming publication is played from the very beginning? I suspect initial negotiations cut some time.

No. LiveKit is a realtime system so there is no buffering of data. What that means is whenever the publisher is pushing out data, if the subscriber is not ready to receive them, they will be missed.

I suspect that's what you are seeing.. the publisher is playing out audio "too early", in the sense that the subscriber has yet to set up the subscription of that track.

We are making this a bit easier moving forward, with a new notification to the publisher that a newly published track has been subscribed to successfully.

**Comment by bigmoveus** [comment]
I I have same problem just when new version will be ready that improved this subject ?

**Comment by denis-obukhov** [comment]
@davidzhao Is there any update for this: 

> We are making this a bit easier moving forward, with a new notification to the publisher that a newly published track has been subscribed to successfully.

**Comment by bkbasavaraju** [comment]
Even we are facing the same issue. We upgraded the LiveKit server to latest 1.8.4 version and the issue exists. In our case, we are using LiveKit for the OpenAI Realtime Voice and Audio response from the AI is getting trimmed at the starting of the conversation.
Is there any workaround?

**Comment by lijoabraham** [comment]
+1, we are also facing this issue and it is an intermittent issue where initial part of the text is missing in audio and we can see the output text is coming correctly in the logs

**Comment by yu-supplication** [comment]
Updating the LiveKit server to the latest version 1.9.3 can solve this problem

**Comment by davidzhao** [comment]
this event has been added awhile ago. clients would now fire a [LocalTrackSubscribed](https://github.com/livekit/client-sdk-js/blob/main/src/room/events.ts#L349) event when its published track has been subscribed to.

**Comment by lijoabraham** [comment]
@davidzhao Is this change available also in python SDK?
